src/Fhir.Anonymizer.Core/Extensions/ElementNodeOperationExtensions.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Fhir.Anonymizer.Core.AnonymizationConfigurations;
+using Fhir.Anonymizer.Core.Models;
+using Fhir.Anonymizer.Core.Processors;
+using Fhir.Anonymizer.Core.Visitors;
+using Hl7.Fhir.ElementModel;
+using Hl7.Fhir.Model;
+using Hl7.Fhir.Specification;
+
+namespace Fhir.Anonymizer.Core.Extensions
+{
+    public static class ElementNodeOperationExtensions
+    {
+        private static readonly PocoStructureDefinitionSummaryProvider s_provider = new PocoStructureDefinitionSummaryProvider();
+
+        public static ElementNode Anonymize(this ElementNode node, AnonymizationFhirPathRule[] rules, Dictionary<string, IAnonymizerProcessor> processors)
+        {
+            AnonymizationVisitor visitor = new AnonymizationVisitor(rules, processors);
+            node.Accept(visitor);
+            node.RemoveNullChildren();
+
+            return node;
+        }
+
+        // Remove null children of current node, and return true => current node is null
+        public static bool RemoveNullChildren(this ElementNode node)
+        {
+            if (node == null)
+            {
+                return true;
+            }
+
+            var children = node.Children().Cast<ElementNode>().ToList();
+            foreach (var child in children)
+            {
+                // Remove child if it is null => return true
+                if (RemoveNullChildren(child))
+                {
+                    node.Remove(child);
+                }
+            }
+
+            bool currentNodeIsEmpty = !node.Children().Any() && node.Value == null;
+            bool currentNodeIsFhirResource = node.IsFhirResource();
+            if (currentNodeIsEmpty && !currentNodeIsFhirResource)
```

Comment:
Curiously, what is the behavior if we add a rule "Patient:redact"? What is the expected of a Patient json, a Patient in Bundle or a contained Patient? Is it an empty json with "{}"?

Reply:
Some unmidified field should not be redact like resource type.

Reply:
So rules like "Patient":"redact" is not supported/allowed?

Reply:
the result is like:

{"resourceType":"Patient","meta":{"security":[{"system":"http://terminology.hl7.org/CodeSystem/v3-ObservationValue","code":"REDACTED","display":"part of the resource is removed"}]}}